source/common/upstream/eds.cc

```diff
@@ -22,16 +24,21 @@ EdsClusterImpl::EdsClusterImpl(const Json::Object& config, Runtime::Loader& runt
 
 void EdsClusterImpl::initialize() { subscription_->start({cluster_name_}, *this); }
 
-bool EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
+void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
   std::vector<HostSharedPtr> new_hosts;
   if (resources.size() != 1) {
-    ENVOY_LOG(warn, "Unexpected EDS resource length: {}", resources.size());
-    return false;
+    // We need to allow server startup to continue, even if we have a bad
+    // config.
+    runInitializeCallbackIfAny();
```

Comment:
Yes I think you would want some type of catch-all that is exactly like onFetchComplete() (which runs no matter what happens, see existing code in RestAPIFetcher).